core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1714,6 +1715,19 @@ public ServerBuilder setHeaders(
      * @see RequestContext#id()
      */
     public ServerBuilder requestIdGenerator(Supplier<? extends RequestId> requestIdGenerator) {
+        final Supplier<? extends RequestId> requestIdSupplier = requireNonNull(requestIdGenerator);
+        this.requestIdGenerator = routingContext -> requestIdSupplier.get();
+        return this;
```

Comment:
Should we delegate to `requestIdGenerator(Function)`?
For example:
```java
requireNonNull(requestIdGenerator, "requestIdGenerator");
return requestIdGenerator(routingCtx -> requestIdSupplier.get());
```

Reply:
Sure! I modify it.